### ISSUE TITLE
[Bug]: Ribbon in .NET 6 (was 5) designer can't add a linegroup or anything else to a tab group

### ISSUE THREAD

**Issue by Tape-Worm** [Krypton-Suite/Standard-Toolkit#262]
**Describe the bug**
After adding a group to the tab in the ribbon, the next level of containers fail with an error about abstract classes.

**To Reproduce**
1. Add a tab (using the properties pane)
2. Add a group
3. Errors on next container level

**Expected behavior**
Should allow adding of a line group, or other container types.

**Screenshots**
Screen shot available here:
https://github.com/Krypton-Suite/Standard-Toolkit/issues/260#issuecomment-897755055

**Comment by Smurf-IV** [comment]
Check with #178

**Comment by Tape-Worm** [comment]
FYI, this problem is getting very difficult to manage. 

The workaround provided only works for simple projects. Anything remotely complex (e.g. with custom controls embedded into the form) starts becoming difficult, at best, and impossible at worst, to manage.

Since I'm writing an API that allows users to write their own plug ins for this application this issue, makes it so others can't use my stuff without implementing a rather annoying hack to make it work which can require some intimate knowledge of how all this stuff works under the hood.

Has there been any progress on getting this issue fixed?  Would you like me to do some testing for you guys on this?

**Comment by Smurf-IV** [comment]
> Has there been any progress on getting this issue fixed?

It is really annoying - I agree.
I'm still battling with this, and it seems that it has got worse now that VS16.11 has released, as NO controls show up !
As linked- check the progress against #178 and the others linked to it.. 

>  implementing a rather annoying hack to make it work

Can you elaborate on your workaround ?

**Comment by Tape-Worm** [comment]
It's just the standard workaround where you make the project target multiple frameworks, one for 4.8 and the other for 5.0. The other workaround where you link to a control/form in a separate .NET 4.8 project is even worse as soon as you add a single custom control and/or embedded resources.

Within my project, the first workaround works fine, but when I try to use it outside of my project source code that's when it all breaks down. Because I do not ship out the 4.8 code, users cannot link against my APIs and they cannot access the ribbon and create their own ribbon interface using .net 5 - which is something that I really want to expose.

I also tried a code-first approach where I started to create a ribbon builder, and that looked promising until I found that I could not access the controls by name via the control hierarchy in the ribbon. 

As for why I don't ship out the 4.8 projects: I have a lot of C# 9 code, and there's parts (e.g. major parts of the Record type) that just don't work in 4.8 regardless of langversion. I #ifdef those lines of code out, thus making the .NET 4.8 code unusable (but can still compile).

If I had the ability to turn off the 4.8 framework, and the ribbon allowed me to set it up in the .NET 5 designer, then all of this would no longer be an issue.

**Comment by Smurf-IV** [comment]
Thanks for the report.. I'll keep pushing to try to fix each time I do a PR.

**Comment by Tape-Worm** [comment]
Thank you

**Comment by Smurf-IV** [comment]
So following the "Manual" method of adding via the properties, and then adding into each collection (Rather than clicking in the design area and performing add).
It (Canary V2108.1) eventually spits out this:
![image](https://user-images.githubusercontent.com/2418812/133479274-a67b9920-ac56-45b9-928f-ebb4e14c6120.png)

But the Attached Debugger to visual studio didn;t even blink, so this is a reflection mess !

**Comment by Tape-Worm** [comment]
Yep, that's what I've been getting. I could be wrong, but that designer looks like it's a standard .NET designer? That could be why you're not seeing anything in the debugger?

**Comment by sdreb3421** [comment]
Sorry for the sidebar question. I'm preparing to upgrade a large net48 project to NET5. I'm hesitating since it uses krypton controls extensively throughout the application. I've been digging through the open issues to try and piece together any issues I might expect, but it's hard to determine what's remaining. Other than the issue presented above, is the next stable build of krypton expected to have any remaining NET5 designer or runtime bugs? I'm fine with targeting multiple frameworks if necessary. Or would it be safer to wait until next year? Thanks

**Comment by Smurf-IV** [comment]
In Visual Studio 17P7 it's even worse with just .net5 :
![image](https://user-images.githubusercontent.com/2418812/139573856-2b7ef31c-37a5-4073-845a-98474ddbf121.png)

**Comment by Smurf-IV** [comment]
But the designers can be accessed via the Collection properties:
![image](https://user-images.githubusercontent.com/2418812/139573939-4970b5f2-1c20-4340-8406-cafecfea9f36.png)

**Comment by Smurf-IV** [comment]
Also the Toolbox does not list "Any" krypton components!

**Comment by Tape-Worm** [comment]
Just checking in on this to see if there's been any progress on fixing the designer issue?  Will you need the new Windows Forms SDK to make this work?

**Comment by Wagnerp** [comment]
Hi @Tape-Worm 

A discussion has been posted here https://github.com/Krypton-Suite/Standard-Toolkit/discussions/684

**Comment by Wagnerp** [comment]
Hi @Tape-Worm 

Yes the new Windows Forms SDK will be needed to fix this, since Microsoft broke something in-between 4.8 and .NET Core. One solution that may work is to add 'net48' to your 'TargetFrameworks'. This will force the designer to use a previous version.

**Comment by Smurf-IV** [comment]
I know this started with ".net5 ...."
But [".NET 5.0 will reach End of Support on May 08, 2022](https://devblogs.microsoft.com/dotnet/dotnet-5-end-of-support-update/)"
So I have changed the title to .Net6 to keep it relevant (And I'm still checking any solution !!)

**Comment by Smurf-IV** [comment]
https://github.com/dotnet/winforms/discussions/7073

**Comment by jfriborg** [comment]
Hi. Any updates on this issue?
I'm experiencing the same error, ...because it is an abstract class, when trying to add an item to a ribbon group.
I'm using VS 2022, .NET 6 and v65.22.6.152 of the Krypton.Ribbon package.
I'm not able to use the suggested workaround with multiple target frameworks or just net48 due to some 3rd part components i have to use...

**Comment by giduac** [comment]
Not really, everyone is waiting for this.

https://github.com/Krypton-Suite/Standard-Toolkit/discussions/684

**Comment by Wagnerp** [comment]
Hi @jfriborg 

Can't really fix until the WinForms designer is properly fixed for .NET.

**Comment by Wagnerp** [comment]
@Smurf-IV Does this need to be a new feature request to add/fix designers?

**Comment by Smurf-IV** [comment]
I think this was supposed to be left open waiting for the MS, then it would be fixed against this.
There are other "bugs" in the backlog which also refer to designer issues, so they should all be taken into account.

**Comment by Wagnerp** [comment]
> I think this was supposed to be left open waiting for the MS, then it would be fixed against this. There are other "bugs" in the backlog which also refer to designer issues, so they should all be taken into account.

@Smurf-IV Great, if you come across any bug reports relating to this, please apply the 'designer' tag if not already applied

**Comment by Smurf-IV** [comment]
To summarize: (Tested in VS 17.7.4, and using the 80.23.8.233 canary build)
- IFF .net 6 onwards is the only framework type then the experience is as per https://github.com/Krypton-Suite/Standard-Toolkit/issues/262#issuecomment-955655525
- Using a mix of .net48 and .net6 gives **NOTHING** in the designer !!
- Using just .net48 **NOTHING** in the designer !!

The following exception is being thrown:
 `Microsoft.DotNet.DesignTools.Client.DesignToolsServerException: Type  could not be read from the data in line 167, position 5.  The type's internal structure may have changed.  Either implement ISerializable on the type or provide a type converter that can provide a more reliable conversion format, such as text or an array of bytes.  The conversion exception was: Unable to find assembly 'Krypton.Toolkit, Culture=neutral, PublicKeyToken=a87e673e9ecb6e8e'. ---> Microsoft.DotNet.DesignTools.Client.DesignToolsServerException: Type  could not be read from the data in line 167, position 5.  The type's internal structure may have changed.  Either implement ISerializable on the type or provide a type converter that can provide a more reliable conversion format, such as text or an array of bytes.  The conversion exception was: Unable to find assembly 'Krypton.Toolkit, Culture=neutral, PublicKeyToken=a87e673e9ecb6e8e'. Line 167, position 5. ---> Microsoft.DotNet.DesignTools.Client.DesignToolsServerException: Unable to find assembly 'Krypton.Toolkit, Culture=neutral, PublicKeyToken=a87e673e9ecb6e8e'.`

**Comment by giduac** [comment]
Hi @Tape-Worm, @Smurf-IV & @Wagnerp 

What is the status of this ticket?

**Comment by Tape-Worm** [comment]
@giduac 

It's been a long time since I last looked at this (probably 6 months to a year), but the last I checked it was still broken. From what I understood wagner and smurf were waiting on some sort of thing from Microsoft to aid in building custom components with designers (I know very little about it I'm afraid).

**Comment by giduac** [comment]
@PWagner1 & @Smurf-IV 

If this is connected the Winforms OOP designer does this ticket still need to be open?

**Comment by Tape-Worm** [comment]
You can close this. I've decided to no longer use Krypton in my projects.

**Comment by PWagner1** [comment]
@giduac 

Yes it's connected to OOP
